include/deal.II/base/hdf5.h

```diff
@@ -0,0 +1,1073 @@
+
+  public:
+    /**
+     * Reads data of the dataset.
+     *
+     * Datatype conversion takes place at the time of the read operation and is
+     * automatic. See the <a
+     * href="https://support.hdfgroup.org/HDF5/doc/UG/HDF5_Users_Guide-Responsive%20HTML5/index.html#t=HDF5_Users_Guide%2FDatatypes%2FHDF5_Datatypes.htm%23TOC_6_10_Data_Transferbc-26&rhtocid=6.5_2">Data
+     * Transfer: Datatype Conversion and Selection</a>  section in the HDF5
+     * User's Guide.
+     *
+     * `Container` can be `std::vector<float>`, `std::vector<double>`,
+     * `std::vector<std::complex<float>>`, `std::vector<std::complex<double>>`,
+     * `std::vector<int>`, `std::vector<unsigned int>`, `Vector<float>`,
+     * `Vector<double>`, `Vector<std::complex<float>>`,
+     * `Vector<std::complex<double>>`, `FullMatrix<float>`,
+     * `FullMatrix<double>`, `FullMatrix<std::complex<float>>` or
+     * `FullMatrix<std::complex<double>>`.
+     */
+    template <typename Container>
+    Container
+    read();
+
+    /**
+     * Reads data of a subset of the dataset.
+     *
+     * Datatype conversion takes place at the time of the read operation and is
+     * automatic. See the <a
+     * href="https://support.hdfgroup.org/HDF5/doc/UG/HDF5_Users_Guide-Responsive%20HTML5/index.html#t=HDF5_Users_Guide%2FDatatypes%2FHDF5_Datatypes.htm%23TOC_6_10_Data_Transferbc-26&rhtocid=6.5_2">Data
+     * Transfer: Datatype Conversion and Selection</a>  section in the HDF5
+     * User's Guide.
+     *
+     * The selected elements can be scattered and take any shape in the dataset.
+     * For example, in the case of a dataset with rank 4 a selection of 3 points
+     * will be described by a 3-by-4 array. Note the indexing is zero-based. To
+     * select the points (1,1,1,1), (14,6,12,18), and (8,22,30,22), the point
+     * selection array would be as follows:
+     *
+     * @code
+     *    0  0  0  0
+     *
+     *   13  5 11 17
+     *
+     *    7 21 29 21
+     * @endcode
+     *
+     * <a
+     * href="https://support.hdfgroup.org/newsletters/newsletter140.html">Parallel
+     * HDF5 supports collective I/O on point selections.</a>
+     *
+     * Datatype conversion takes place at the time of the read operation and is
+     * automatic. See the <a
+     * href="https://support.hdfgroup.org/HDF5/doc/UG/HDF5_Users_Guide-Responsive%20HTML5/index.html#t=HDF5_Users_Guide%2FDatatypes%2FHDF5_Datatypes.htm%23TOC_6_10_Data_Transferbc-26&rhtocid=6.5_2">Data
+     * Transfer: Datatype Conversion and Selection</a>  section in the HDF5
+     * User's Guide.
+     */
+    template <typename Container>
+    Container
+    read_selection(const std::vector<hsize_t> &coordinates);
+
+    /**
+     * Reads a hyperslab from the dataset. The parameters are summarized
+     * below:
+     *  - Offset: The starting location for the hyperslab.
+     *  - Count: The number of elements to select along each dimension.
+     *
+     * Stride and block are set to NULL.
```

Comment:
what does this refer to? Internal parameters of the call to HDF5? If so, I would phrase this as follows to make that clear: 
```
When reading a hyperslab, HDF5 also allows to provide "stride" and "block" parameters (see
the HDF5 documentation). These are not used by the current function and set to `NULL`.
```

Reply:
But you could point at the next function in this context.